File: fsharp/step0_repl.fs
module REPL
    let read input =
        input

    let eval ast =
        ast

    let print v =
        printfn "=> %A" v

    let rep input =
        input
        |> read
        |> eval
        |> print

    [<EntryPoint>]
    let rec main args =
        match System.Console.ReadLine() with
        | null -> 0
        | input -> 
            rep input
            main args

File: impls/myjava/app/src/main/java/mal/step0_repl/Step0_REPL.java
package mal.step0_repl;


public class Step0_REPL {

    private static String READ(String string) {
        return string;
    }

    private static String EVAL(String string) {
        return string;
    }

    private static String PRINT(String string) {
        return string;
    }

    public static String rep(String input) {
        return PRINT(EVAL(READ(input)));
    }

}
